drivers/adc/adc_stm32.c

```diff
@@ -758,71 +778,81 @@ static int start_read(const struct device *dev,
 		return -EINVAL;
 	}
 
-	uint32_t channels = sequence->channels;
-	uint8_t index = find_lsb_set(channels) - 1;
-
-	if (channels > BIT(index)) {
-		LOG_ERR("Only single channel supported");
-		return -ENOTSUP;
-	}
+	data->buffer = sequence->buffer;
+	data->channels = sequence->channels;
+	data->channel_count = 0;
+	data->samples_count = 0;
 
-	adc_stm32_setup_channels(dev, index);
+	for (uint8_t channel_id = 0; channel_id < STM32_CHANNEL_COUNT; channel_id++) {
+		if (data->channels & BIT(channel_id)) {
```

Comment:
Instead of looping for every bit, you can efficiently only loop on the set bits:
```c
uint8_t channel_id;
for (uint32_t channels = data->channels; channels; channels &= ~BIT(channel_id)) {
        channel_id = find_msb_set(channels) - 1;
}
```

Reply:
Added here and in `adc_stm32_teardown_channels` where it removed an extra indentation. It's a nice trick!